get_modis.py

```diff
@@ -241,91 +265,237 @@ def get_modisfiles(username, password, platform, product, year, tile, proxy,
                                                      "%j/%Y")) for i in
              range(doy_start, doy_end)]
     url = "%s/%s/%s/" % (base_url, platform, product)
-    dates = parse_modis_dates(url, dates, product, out_dir, ruff=ruff)
-    them_urls = []
-    for date in dates:
-        r = requests.get("%s/%s" % (url, date), verify=False)
-        for line in r.text.split("\n"):
-            if line.decode().find(tile) >= 0:
-                if line.decode().find(
-                        ".hdf") >= 0 > line.decode().find(".hdf.xml"):
-                    fname = line.decode().split("href=")[1].split(">")[0].strip('"')
-
-                    if not os.path.exists(os.path.join(out_dir, fname)):
-                        them_urls.append("%s/%s/%s" % (url, date, fname))
-                    else:
-                        if verbose:
-                            LOG.info("File %s already present. Skipping" % fname)
-
-    with requests.Session() as s:
-        s.auth = (username, password)
-        for the_url in them_urls:
-            r1 = s.request('get', the_url)
-            r = s.get(r1.url, stream=True)
-            if not r.ok:
-                raise IOError("Can't start download... [%s]" % fname)
-            file_size = int(r.headers['content-length'])
-            fname = the_url.split("/")[-1]
-            LOG.info("Starting download on %s(%d bytes) ..." %
-                     (os.path.join(out_dir, fname), file_size))
-            with open(os.path.join(out_dir, fname), 'wb') as fp:
-                for chunk in r.iter_content(chunk_size=CHUNKS):
-                    if chunk:
-                        fp.write(chunk)
-                fp.flush()
-                os.fsync(fp)
+    dates_available = None
+    
+    count_reconn_attempts = 0
+    while count_reconn_attempts <= reconnection_attempts:
+        if verbose:
+            LOG.info("Session Attempt %d" % (count_reconn_attempts+1))
+
+        try:
+            with requests.Session() as s:
+                s.mount(base_url, requests.adapters.HTTPAdapter(max_retries=5))
+
+                ## Login to the EarthData Service for this session
```

Comment:
Is this authenticity token needed? I think I just used a session with no issues not that long ago... Or is it needed for the download to work with NSIDC (I seem to remember it wasn't necessary either...)?

Reply:
I wrote this code about three years ago so memory is a bit hazy, but my assumption is that it must have been necessary then otherwise I wouldn't have gone to the trouble of GETing then POSTing the auth token.

Reply:
(the login process is the same whether USGS or NSIDC - they both go through the EarthData login service)